src/custom-elements.js

```diff
@@ -635,6 +646,21 @@ let Deferred;
     }
   }
 
+  /**
+   * @param {!Map<string, !CustomElementDefinition>} definitions
+   * @param {!Node|!HTMLElement|null} node
+   */
+  function getDefinition(definitions, node) {
+    const name = typeof node.is === 'string' ? node.is : node.tagName.toLowerCase();
+    const definition = definitions.get(name);
+    if (definition) {
+      // Make sure local name matches the actual node's tagName
+      return definition.localName === node.tagName.toLowerCase() || definition.localName === node.is ? definition : null;
```

Comment:
You can just use `node.localName` instead of `node.tagName.toLowerCase()`. I believe that will be more accurate in Unicode scenarios as well.